minimint-core/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "minimint-core"
+version = "0.0.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+async-trait = "0.1.42"
+futures = "0.3.9"
+bincode = "1.3.1"
+bitcoin = "0.27.0"
+hex = "0.4.2"
+itertools = "0.10.0"
+minimint-api = { path = "../minimint-api" }
+minimint-derive = { path = "../minimint-derive" }
+minimint-ln = { path = "../modules/minimint-ln" }
+minimint-mint = { path = "../modules/minimint-mint" }
+minimint-wallet = { path = "../modules/minimint-wallet", default-features = false }
```

Comment:
Why did minimint-core need to be refactored out (could you add a description in the arch doc)?

Seems like a cleaner approach instead of having the client code depend on the server code, would be to refactor out a common library for each of the modules that both the server and client side could depend on, although this is a bigger change.

Reply:
`minimint-core` is just a generic name. Main purpose is to expose types from `minimint::outcome` and `minimint::transaction`, and `ClientConfig` to the client without depending on the server.

Maybe `minimint-types` is a better name?